### ISSUE TITLE
How to understand the process of  hyperopt tune

### ISSUE THREAD

**Issue by lyw615** [mlflow/mlflow#1609]
`    For hyperopt tune in examples/hyperparam/search_hyperopt.py, my understanding is to adjust the values of momentum and learning_rate to max-runs times. Each time training the number of 'epochs' times epoch and adjusting the weights in the model during this process so that the loss value is reduced. And finally the max-runs number optimal models are compared to the optimal model.`

**Comment by ankitmathur-db** [comment]
Yes, this script uses hyperopt (http://hyperopt.github.io/hyperopt/) and explores a space of learning rate and momentum to identify the best model. Hyperopt takes a max-runs argument which tells you at most how many runs it will use to identify the best model. Each run is independent and the weights are updated with the optimization strategy at each step over a fixed number of epochs specified in the command line argument. 

Closing since this isn't an issue on the project.
